packages/abstractions/src/requestInformation.ts

```diff
@@ -104,25 +107,95 @@ export class RequestInformation {
     contentType?: string | undefined,
     ...values: T[]
   ): void => {
+    const writer = this.getSerializationWriter(
+      requestAdapter,
+      contentType,
+      values
+    );
+    if (!this.headers) {
+      this.headers = {};
+    }
+    if (values.length === 1) {
+      writer.writeObjectValue(undefined, values[0]);
+    } else {
+      writer.writeCollectionOfObjectValues(undefined, values);
+    }
+    this.setContentAndContentType(writer, contentType);
+  };
+  private setContentAndContentType = (
+    writer: SerializationWriter,
+    contentType?: string | undefined
+  ) => {
+    if (contentType) {
+      this.headers[RequestInformation.contentTypeHeader] = contentType;
+    }
+    this.content = writer.getSerializedContent();
+  };
+  private getSerializationWriter = <T>(
+    requestAdapter?: RequestAdapter | undefined,
+    contentType?: string | undefined,
+    ...values: T[]
+  ): SerializationWriter => {
     if (!requestAdapter) throw new Error("httpCore cannot be undefined");
     if (!contentType) throw new Error("contentType cannot be undefined");
     if (!values || values.length === 0) {
       throw new Error("values cannot be undefined or empty");
     }
-
-    const writer = requestAdapter
+    return requestAdapter
       .getSerializationWriterFactory()
       .getSerializationWriter(contentType);
+  };
+  /**
+   * Sets the request body from a model with the specified content type.
+   * @param values the scalar values to serialize.
+   * @param contentType the content type.
+   * @param requestAdapter The adapter service to get the serialization writer from.
+   * @typeParam T the model type.
+   */
+  public setContentFromScalar = <T>(
```

Comment:
Please add unit tests validating this change.

Reply:
added